src/Expr.cc

```diff
@@ -1249,6 +1248,16 @@ void BinaryExpr::PromoteForInterval(ExprPtr& op)
 		op = make_intrusive<ArithCoerceExpr>(op, TYPE_DOUBLE);
 	}
 
+bool BinaryExpr::IsScalarVecOp()
```

Comment:
How about renaming to `IsScalableAggregateOp()` because the is now broader than just vectors?